.jenkins.groovy

```diff
@@ -0,0 +1,186 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+def deployableBranch = env.BRANCH_NAME ==~ /(1.7.x|1.8.x|main)/
+
+pipeline {
+
+    agent none
+
+    options {
+        // When we have test-fails e.g. we don't need to run the remaining steps
+        skipStagesAfterUnstable()
+        buildDiscarder(logRotator(numToKeepStr: '5', artifactNumToKeepStr: '5'))
+    }
+
+    stages {
+        stage('Build') {
+            matrix {
+                axes {
+                    axis {
+                        // https://cwiki.apache.org/confluence/display/INFRA/JDK+Installation+Matrix
+                        name 'MATRIX_JDK'
+                        values 'jdk_1.8_latest',
+                                'jdk_11_latest',
+                                'jdk_14_latest'
+                    }
+                    // Additional axess, like OS and maven version can be configured here.
+                }
+
+                agent {
+                    node {
+                        // https://cwiki.apache.org/confluence/display/INFRA/ci-builds.apache.org
+                        label 'ubuntu'
+                    }
+                }
+
+                options {
+                    // Configure an overall timeout for the build of one hour.
+                    timeout(time: 1, unit: 'HOURS')
+                }
+
+                tools {
+                    // https://cwiki.apache.org/confluence/display/INFRA/Maven+Installation+Matrix
+                    maven 'maven_3_latest'
+                    jdk "${MATRIX_JDK}"
+                }
+
+                stages {
+                    stage('Initialization') {
+                        steps {
+                            echo 'Building Branch: ' + env.BRANCH_NAME
+                            echo 'Using PATH = ' + env.PATH
+                        }
+                    }
+
+                    stage('Cleanup') {
+                        steps {
+                            echo 'Cleaning up the workspace'
+                            cleanWs()
+                        }
+                    }
+
+                    stage('Checkout') {
+                        steps {
+                            echo 'Checking out branch ' + env.BRANCH_NAME
+                            checkout scm
+                        }
+                    }
+
+                    stage('License check') {
+                        steps {
+                            echo 'License check'
+                            sh 'mvn --batch-mode -Drat.consoleOutput=true apache-rat:check'
+                        }
+                    }
+
+                    stage('Build') {
+                        steps {
+                            echo 'Building'
+                            sh 'mvn --update-snapshots --batch-mode --errors clean verify -Pdocs -Dmaven.test.failure.ignore=true'
```

Comment:
We shouldn't be using snapshots, so updating should not be needed.

Reply:
Fixed :)